syntax = "proto3";

option go_package = "grpc.models";

package ficus;

message GrpcGuid {
  string guid = 1;
}

message GrpcColor {
  uint32 red = 1;
  uint32 green = 2;
  uint32 blue = 3;
}

message GrpcUuid {
  string uuid = 1;
}

message GrpcStringKeyValue {
  string key = 1;
  string value = 2;
}

message GrpcTimeSpan {
  uint64 nanoseconds = 1;
}

message GrpcDateTime {
  uint64 nanosSinceUnixEpoch = 1;
}